script.js

```diff
@@ -0,0 +1,18 @@
+const Input = document.getElementById('user-input');
+const check = document.getElementById('check-btn');
+const clear = document.getElementById('clear-btn');
+const results = document.getElementById('results-div');
+check.addEventListener('click', () => {
+  const regex = /^(1\s?)?(\(\d{3}\)|\d{3})([\s-]?)\d{3}([\s-]?)\d{4}$/;
+  if (!Input.value) {
+    alert('Please provide a phone number');
+  } else if (regex.test(Input.value)) {
+    results.innerText = `Valid US number: ${Input.value}`;
+  } else {
+    results.innerText = `Invalid US number: ${Input.value}`;
+  }
+});
+clear.addEventListener('click', () => {
+  results.textContent = '';
+  Input.value = '';
+});
```

Comment:
- [OPTIONAL] - Try to add your Event Listeners in while DOM content is loading, it helps to add all of your event listeners to be ready to the action right when your page loads and prevents any side effects below is an example to help you do that:

```javascript
document.addEventListener('DOMContentLoaded', () => {
  //In here add all of your addEventListener functions
  // don't forget to add the constants that you need in your event listeners here as the scope of this one is higher than other codes
  const Input = document.getElementById('user-input');
  const check = document.getElementById('check-btn');
  const clear = document.getElementById('clear-btn');
  const results = document.getElementById('results-div');
  check.addEventListener('click', () => {
    const regex = /^(1\s?)?(\(\d{3}\)|\d{3})([\s-]?)\d{3}([\s-]?)\d{4}$/;
    if (!Input.value) {
      alert('Please provide a phone number');
    } else if (regex.test(Input.value)) {
      results.innerText = `Valid US number: ${Input.value}`;
    } else {
      results.innerText = `Invalid US number: ${Input.value}`;
    }
  });
  clear.addEventListener('click', () => {
    results.textContent = '';
    Input.value = '';
  });
});
```

Reply:
@ahzamir thanks for your reviewing . I added an Event Listeners in while DOM content is loading. in my project

Reply:
@FarwaMuhibZada 

- Well done nothing to say other than you can merge your branch. 🎉👍✅